src/catalog/catalog_manager.py

```diff
@@ -37,89 +37,145 @@ def __new__(cls):
         return cls._instance
 
     def bootstrap_catalog(self):
+        """Bootstraps catalog.
 
-        # eva_dir = ConfigurationManager().get_value("core", "location")
-        # output_url = os.path.join(eva_dir, CATALOG_DIR)
-        # LoggingManager().log("Bootstrapping catalog" + str(output_url),
-        #                      LoggingLevel.INFO)
+        This method runs all tasks required for using catalog. Currently,
+        it includes only one task ie. initializing database. It creates the
+        catalog database and tables if they do not exist.
+
+        """
         LoggingManager().log("Bootstrapping catalog", LoggingLevel.INFO)
         init_db()
-        # # Construct output location
-        # catalog_dir_url = os.path.join(eva_dir, "catalog")
-        #
-        # # Get filesystem path
-        # catalog_os_path = urlparse(catalog_dir_url).path
-        #
-        # # Check if catalog exists
-        # if os.path.exists(catalog_os_path):
-        #     # Load catalog if it exists
-        #     load_catalog_dataframes(catalog_dir_url,
-        #     self._catalog_dictionary)
-        # else:
-        #     # Create catalog if it does not exist
-        #     create_catalog_dataframes(
-        #         catalog_dir_url, self._catalog_dictionary)
+
+    def create_metadata(self, name: str, file_url: str,
```

Comment:
can we refer to this as 'create_dataset'?

Reply:
can't `file_url`be automatically generated?

Reply:
I don't think init_db should be called here. 

I feel it's a bad idea to run the migrations when the class instance is being created. This should be script/command outside the catalog. Similar to the **initdb** command of Postgres.